framework/base/Module.php

```diff
@@ -283,6 +300,58 @@ public function setLayoutPath($path)
     }
 
     /**
+     * Returns current module version.
+     * If version is not explicitly set, [[defaultVersion()]] method will be used to determine its value.
+     * @return integer|string current module version.
+     * @since 2.0.11
+     */
+    public function getVersion()
+    {
+        if ($this->_version === null) {
+            $this->_version = $this->defaultVersion();
+        } else {
+            if (!is_scalar($this->_version)) {
+                $this->_version = call_user_func($this->_version, $this);
+            }
+        }
+        return $this->_version;
+    }
+
+    /**
+     * Sets current module version.
+     * @param integer|string|callable $version version specification.
+     * Version can be specified as a PHP callback, which can accept module instance as an argument and should
+     * return the actual version. For example:
+     *
+     * ```php
+     * function (Module $module) {
+     *     //return string|integer
+     * }
+     * ```
+     *
+     * @since 2.0.11
+     */
+    public function setVersion($version)
+    {
+        $this->_version = $version;
+    }
+
+    /**
+     * Returns default module [[version]] based last modification time of the [[basePath]].
+     * Child class may override this method to provide more specific version detection.
+     * @return string|integer module version.
+     * @since 2.0.11
+     */
+    protected function defaultVersion()
+    {
+        $versionTimestamp = @filemtime($this->getBasePath());
```

Comment:
> current implementation will pick up last modification date from runtime directory as well

Did you test it? Modifications in `runtime` should not change mtime of root directory.